hello-robot/stretch_body
# Velocity control for dxl joints + bugfixes

This PR introduces the `set_velocity()` API to the Dynamixel joints, including the head pan/tilt and wrist yaw/pitch/roll/gripper. This is the same velocity control API used by the other joints/mobile base. Additionally, this PR fixes a bug where interrupting a low-level communication with the Dxls (e.g. through a SIGINT) would cause the dxl to become inaccessible until the Linux process was killed.

An example script using the new API:
```python
import stretch_body.robot

r = stretch_body.robot.Robot()
r.startup()
assert(r.is_calibrated()) # the robot must be homed

# move the wrist_yaw counter clockwise at 0.1 rad/s
r.end_of_arm.get_joint('wrist_yaw').set_velocity(0.1)
```

How to test
----------
 1. Install this branch locally using [the instructions here](https://github.com/hello-robot/stretch_body/blob/master/body/README.md#developing).
 2. Run the example script above and verify the wrist yaw joint rotates at 0.1 rad/s
